java/ql/lib/semmle/code/java/dataflow/TypeFlow.qll

```diff
@@ -8,757 +8,359 @@
  * explicit or implicit cast that lost type information.
  */
 
-import java
+import java as J
 private import semmle.code.java.dispatch.VirtualDispatch
 private import semmle.code.java.dataflow.internal.BaseSSA
 private import semmle.code.java.controlflow.Guards
-
-private newtype TTypeFlowNode =
-  TField(Field f) { not f.getType() instanceof PrimitiveType } or
-  TSsa(BaseSsaVariable ssa) { not ssa.getSourceVariable().getType() instanceof PrimitiveType } or
-  TExpr(Expr e) or
-  TMethod(Method m) { not m.getReturnType() instanceof PrimitiveType }
-
-/**
- * A `Field`, `BaseSsaVariable`, `Expr`, or `Method`.
- */
-private class TypeFlowNode extends TTypeFlowNode {
-  string toString() {
-    result = this.asField().toString() or
-    result = this.asSsa().toString() or
-    result = this.asExpr().toString() or
-    result = this.asMethod().toString()
+private import codeql.typeflow.TypeFlow
+
+private module Input implements TypeFlowInput<J::Location> {
+  private newtype TTypeFlowNode =
+    TField(Field f) { not f.getType() instanceof PrimitiveType } or
+    TSsa(BaseSsaVariable ssa) { not ssa.getSourceVariable().getType() instanceof PrimitiveType } or
+    TExpr(Expr e) or
+    TMethod(Method m) { not m.getReturnType() instanceof PrimitiveType }
+
+  /** Gets `t` if it is a `RefType` or the boxed type if `t` is a primitive type. */
+  private RefType boxIfNeeded(J::Type t) {
+    t.(J::PrimitiveType).getBoxedType() = result or
```

Comment:
We make plenty of use of unqualified `import java` elsewhere, including `PrimitiveType`, so might as well simplify here.
```suggestion
    t.(PrimitiveType).getBoxedType() = result or
```

Reply:
You're right. The qualifier is only necessary when talking about `Type`. I've removed the other unnecessary qualifiers in 1775bdee5f20f8c957addfad3d31025ac774546c